apps/allocations/app/components/Panel/NewBudget.js

```diff
@@ -0,0 +1,107 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { Info, TextInput } from '@aragon/ui'
+
+import { Form, FormField } from '../Form'
+import { isStringEmpty } from '../../utils/helpers'
+
+// TODO:: This should be votingTokens from account?
+const INITIAL_STATE = {
+  name: '',
+  nameError: true,
+  amount: '',
+  amountError: true,
+}
+
+class NewBudget extends React.Component {
+  static propTypes = {
+    onCreateBudget: PropTypes.func.isRequired
+  }
+
+  state =  INITIAL_STATE
+
+  changeField = e => {
+    if (e.target.name === 'name') {
+      if (isStringEmpty(e.target.value)) {
+        this.setState({ nameError: true })
+      }
+      else {
+        this.setState({ nameError: false })
+      }
+    }
+    else if (e.target.name === 'amount') {
+      if (isStringEmpty(e.target.value) || e.target.value <= 0) {
+        this.setState({ amountError: true })
+      }
+      else {
+        this.setState({ amountError: false })
+      }
+    }
+    this.setState({ [e.target.name]: e.target.value })
+  }
+
+  createBudget = () => {
+    const { name, amount } = this.state
+
+    this.props.onCreateBudget({
+      name,
+      amount,
+    })
+    this.setState(INITIAL_STATE)
+  }
+
+  render() {
+
+    return (
+      <React.Fragment>
+
+
+        <Form
+          onSubmit={this.createBudget}
+          // heading={this.props.heading}
+          submitText="Create budget"
+          disabled={this.state.nameError || this.state.amountError}
+        >
+          <FormField
+            required
+            label="name"
```

Comment:
Just double-checking: should these labels be title-case? "Name" and "Amount", not "name" and "amount"?

Reply:
> Just double-checking: should these labels be title-case? "Name" and "Amount", not "name" and "amount"?

It does not matter actually, since every label is shown as smallcaps